peach_tuntiwong/week01/day2/js/main.js

```diff
@@ -0,0 +1,101 @@
+var solution = ['F', 'O', 'X'];
+var currentGuess = ['_', '_', '_'];
+
+var checkLetter = function(letter) {
+  var inputLetter = letter.toUpperCase(); // Set all input to uppercase
+  if (inputLetter == 'F') { // Return 1 if guess correct, return 0 if guess wrong
+    currentGuess[0] = inputLetter;
+    return 1;
+  } else if (inputLetter == 'O') {
+    currentGuess[1] = inputLetter;
+    return 1;
+  } else if (inputLetter == 'X') {
+    currentGuess[2] = inputLetter;
+    return 1;
+  } else {
+    return 0;
+  }
+}
+
+var guessLetter = function(letter) {
+  checkLetter(letter);
+  console.log(currentGuess);
+  if (currentGuess.indexOf('_') == -1) { // Check for winning condition
+    console.log('Congratulation, you won!');
```

Comment:
Nice way to test for the end of the game! No counter variables required. On the other hand, `indexOf()` is doing some iteration internally, so it's slightly less efficient than using a counter variable.